src/fides/api/ops/service/connectors/email_connector.py

```diff
@@ -28,202 +24,104 @@
+            action_type=ActionType.erasure
         )
+        return bool(erasure_rules and "email" in user_identities.keys())
 
-        # Raises a special exception just to update the ExecutionLog message.  The email send itself
-        # is postponed until all collections have been visited.
-        raise PrivacyRequestErasureEmailSendRequired("email prepared")
-
-    def build_masking_instructions(
-        self, node: TraversalNode, policy: Policy, input_data: Dict[str, List[Any]]
-    ) -> ManualAction:
-        """
-        Generate information on how to find and mask relevant records on this collection.
+    def send_erasure_email(self, privacy_requests: Query) -> None:
+        identities = set()
+        for privacy_request in privacy_requests:
+            identities.add(privacy_request.get_cached_identity_data().get("email"))
 
-        Because we don't have any rows from a completed "access request" for this source, we pass on both "access"
-        and "erasure" instructions to the source.
-        """
+        db = Session.object_session(self.configuration)
+        email_service: Optional[str] = _get_email_messaging_config_service_type(db=db)
 
-        # Values for locating relevant records on the current collection.
-        locators: Dict[str, Any] = node.typed_filtered_values(input_data)
-        # Add additional locators from the same dataset for which we don't have data
-        for edge in node.incoming_edges_from_same_dataset():
-            append(locators, edge.f2.field_path.string_path, str(edge.f1))
+        try:
+            dispatch_message(
+                db=db,
+                action_type=MessagingActionType.MESSAGE_ERASURE_REQUEST_FULFILLMENT,
+                to_identity=Identity(email=self.config.recipient_email_address),
+                service_type=email_service,
+                message_body_params=ErasureRequestBodyParams(
+                    controller=_get_org_name(db),
+                    third_party_vendor_name=self.config.third_party_vendor_name,
+                    identities=identities,
+                ),
+            )
 
-        # Build which fields to mask and how to mask
-        fields_to_mask_by_rule: Dict[Rule, List[FieldPath]] = self.query_config(
-            node
-        ).build_rule_target_field_paths(policy)
-        mask_map: Dict[str, Any] = {}
-        for rule, field_paths in fields_to_mask_by_rule.items():
-            for rule_field_path in field_paths:
-                # Map field paths to the specified masking strategy.  If multiple rules target
-                # the same field path, the last one will win
-                mask_map[rule_field_path.string_path] = (
-                    rule.masking_strategy.get("strategy")
-                    if rule.masking_strategy
-                    else None
+            # create an audit event for each privacy request ID
+            for privacy_request in privacy_requests:
+                ExecutionLog.create(
```

Comment:
Using `ExcutionLog` instead of `AuditLog` so I can show the connector name in the logs instead of just a generic "Request email_sent" message

Reply:
As part of the consent reporting work we're going to have be more clear as to whether or not we skipped sending an email to a service, so I will at least need to add this kind of thing for the consent email connectors if it's not added as part of this work.

Reply:
Noted this TODO in the consent work